src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -546,9 +560,24 @@ public void validate(FailureCollector failureCollector, Map<String, String> argu
         validateJobLabelKeyValue(failureCollector);
       }
 
+      if (!containsMacro(WRITE_PREFERENCE)) {
+        validateWritePreference(failureCollector, getWritePreference());
+      }
+
       failureCollector.getOrThrowException();
     }
 
+    void validateWritePreference(FailureCollector failureCollector, String writePreference) {
+      Set<String> validPreferences = ImmutableSet.of("WRITE_EMPTY", "WRITE_APPEND", "WRITE_TRUNCATE");
```

Comment:
use `com.google.cloud.bigquery.JobInfo.WriteDisposition.WRITE_APPEND.name()` and so on.